LICENSE.mustache

```diff
@@ -81,7 +81,8 @@ When this license requires you to contribute software:
 
 2.  Ensure that each part of that source code is available
     to the public under a license at least as permissive
-    as this one.
+    as this one, such as MIT, two-clause BSD license,
+    Apache 2.0, Charity 3.0.0, or Blue Oak Model.
```

Comment:
Should this also mention the GPL?

Reply:
@zkat, definitely crossed my mind! Unfortunately, like everything touched by the GPL, it's complicated, on a few levels.

All the GPLs are filled with language that make even sections with fairly simple intent complicated. And GPLv3 far moreso than GPLv2. For example, I would call the anti-Tivoization, liberty-or-death, and interactive-notice rules _less_ permissive than Parity, in each of their respects.

Rather than get into yet another big, inevitable, unresolvable debate about what the GPLs mean, plus a big debate about what "permissive" means, I suppose we could just say that GPLs qualify, whether they're as permissive or not. But frankly, I think that's probably more trouble than it's worth. If passing around copies of the complete program involves both GPL- and Parity-licensed code, the one distributing can't make the whole available under GPL. It's still an unresolvable license-compatibility problem, even though Parity offers flexibility to avoid compatibility issues, because GPL doesn't do the same for Parity.

That's why I'm so loathe to use the term "license compatibility". When we're talking two copyleft licenses, rather that GPL and some permissive license, it's not a question of whether the other copyleft license is "GPL compatible". It's really a question of whether the other copyleft license yields to GPL.

GPL is "my way or the highway", even if the other way is stronger copyleft.

Reply:
Good points. The one argument for GPL I can think of is that as a developer who is using Parity, I could then say: "Well, really not my fault you can't use this, talk to the developer of the GPL'ed library". Or, if whoever complains uses GPL, they are responsible themselves (and maybe there is even a small chance they'd move over to Parity themselves).  If the situation is unclear because there's a discussion whether Parity is more or less permissive, that would not be possible.

This flexibility in terms of 'compatibility' is one of the great strengths/innovations of Parity, in my eyes. The GPL issue is definitely one that'll come up sooner or later, either as a real problem, or as a 'quick' argument against Parity in a Reddit/HN-comment, which makes immediate sense to everyone who doesn't know better, or does not like new licenses or copyleft in the first place. Adding GPL/AGPL explicitely as 'alowed' could protect against that to some extent.

Not really sure if this would be worth it though. As you said, in practice it probably doesn't change much.

Reply:
@makkus: To be clear, even if Parity "whitelists" (A)GPL, then (A)GPL will still not whitelist Parity. In order to work, both licenses need to make exception for the other.

In case you haven't noticed before, GPLv3 and AGPLv3 do exactly this in their respective section 13s.

Reply:
Yes, I guess what I was trying to say is that if Parity would whitelist (A)GPL, it'd be (A)GPL's 'fault' if a 3rd party could not use 2 specific libraries together in a project (one Parity-, one (A)GPL-licensed).  Or maybe I'm wrong about that?

As I said, I don't know whether that alone would be worth it, it's just an argument for explicitely whitelisting (A)GPL. So that, when it comes up in an argument, one can say: "Parity would allow for that use-case, it's GPL that prevents you from doing this..."

Maybe I did misunderstand what this is about though, in that case just ignore me :-)

Reply:
@makkus, you haven't understood. And your proposal makes sense.

I guess I'm just not too enthused about putting what is essentially dead code in the license. Whitelisting (A)GPL from Parity would be like building half a bridge across a river, knowing full well that the town on the other side will never build their half.

Reply:
Yes, I totally understand and I don't have strong feelings either way. 

One case that would actually be supported that wouldn't be otherwise would be if soneone would write (A)GPL code and would want to distribute that. They could include Parity code in their distribution, right? Wouldn't make much sense for a library, but if it's an application it'd be allowed, or am I missing something?

Reply:
@makkus, I _don't_ think folks could freely mix Parity and GPL code in their distribution. The GPL requires that the whole be redistributed under GPL terms. Parity does not allow others to change the license terms for the code from Parity to GPL.

Reply:
Oh, I see. Looks like I misunderstood how that works in regards to the GPL from the beginning. In that case, I totally agree, and there is not much point adding an exception at all.

Reply:
Sorry to keep beating this horse, but I just had another thought: aren't we forgetting '...develop, deploy, monitor, or run...'? It's always ok to use GPL'ed software as long as there is no linking involved, right? So in the case of a linter or similar, GPL'ed code wouldn't care if you'd touch it with a Parity licensed dev tool, but depending on whether GPL is considered more or less permissive, this use-case would or wouldn't be permitted by the Parity license. Same thing if I call a GPL'ed application from Parity code via its command-line interface. There are other scenarios I could imagine with similar effects.

I feel like I'm probably missing something again. This stuff does my head in...

Reply:
@makkus, you have a point.

GPL isn't as simple as "if there's no linking, there's no copyleft". But when Parity copyleft kicks in because of, say, use of a dev tool, the code on which the dev tool was run _could_ be distributed under GPL.

I need to think about this a bit.